gramps/gen/lib/grampstype.py

```diff
@@ -151,11 +158,18 @@ def set_object_state(self, attr_dict):
         else:
             self.__string = ""
 
-    def __set_dict(self, value):
+    @singledispatchmethod
+    def set(self, value):
+        self.__value = self._DEFAULT
+        self.__string = ""
+
+    @set.register
+    def set_dict(self, value: dict):
```

Comment:
This should be private.

Reply:
I had not expected it to work if I made `set_dict` private but having tested it, it does.
I'll make the it private

Reply:
This has been addressed in the latest commit